src/components/inputs/uploadJSON.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ChangeEvent, useRef, useState } from 'react';
+
+import CheckIcon from '@mui/icons-material/Check';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import FormControl from '@mui/material/FormControl';
+import FormHelperText from '@mui/material/FormHelperText';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { useFormContext, Controller } from 'react-hook-form';
+
+import { getFieldHelperText } from '../../utils';
+
+/**
+ *
+ * @public
+ */
+export type UploadJSONProps = {
+  name?: string;
+  helperText?: string;
+};
+
+/**
+ * Button to upload JSON and assign it to a form value.
+ *
+ * @public
+ */
+export function UploadJSON({
+  name = 'service_account_key_json',
+  helperText = 'See sidebar for details on locating your Service Account JSON file.',
```

Comment:
You sure those default values are general and could be potentially reused by different plugins?

Reply:
they are valid across the 3 GCP based plugins I've seen so far, but I did want to leave them in a position to be overridden